### PR TITLE
[muxorch] Fixing updateRoute logic

### PR DESCRIPTION
**Pull request by Ndancejic**
What I did:
- Removed logic that prevented routes from updating if an active nexthop was already found in updateRoute cache.
- added test to check for scenario where ecmp route update triggers after updateRoute programs route to point to single nh
- Added logs to some key paths in updateRoute logic to add visibility

Why I did it:
Bug in logic was causing routes getting updated multiple times to skip updateRoute logic

How I did it:
removing logic that checked cache for active nexthops

### CODE REVIEW

File: tests/test_mux.py [hunk 1]
```diff
@@ -584,14 +584,14 @@ def multi_nexthop_test(self, dvs, dvs_route, asicdb, appdb, route, neighbors, ma
                         print("setting %s to %s" % (mux_ports[toggle_index], ACTIVE))
                         self.set_mux_state(appdb, mux_ports[toggle_index], ACTIVE)
                         if start[keep_index] == ACTIVE:
-                            self.check_route_nexthop(dvs_route, asicdb, route, neighbors[keep_index])
+                            self.check_route_nexthop(dvs_route, asicdb, route, neighbors[0])
```
Comment (prsunny) [inline tests/test_mux.py:587]: Why is existing test changed?

Comment (Ndancejic) [reply tests/test_mux.py:587]: The logic changed, so this test was failing. Each time we update the route, we'll be programming the route to point to the first active neighbor. Before this, we were keeping whatever nexthop was previously programmed.

**Comment by prsunny** [comment]
@Ndancejic , can you rebase with the latest 202205 and update the test?

**Review by prsunny — APPROVED** [review]

**Comment by StormLiangMS** [comment]
Hi @Ndancejic could you update ADO and test result for 202305?

**Comment by prsunny** [comment]
> Hi @Ndancejic could you update ADO and test result for 202305?

@StormLiangMS , updated ADO. It is tested and validated on 202205.

**Comment by StormLiangMS** [comment]
> > Hi @Ndancejic could you update ADO and test result for 202305?
> 
> @StormLiangMS , updated ADO. It is tested and validated on 202205.

hi @prsunny typo? 202305 or 202205?

**Comment by StormLiangMS** [comment]
@Ndancejic  @prsunny  suppose typo, just merged.

**Comment by StormLiangMS** [comment]
hi @yxieca for 202311 cherry pick.

**Comment by Ndancejic** [comment]
https://github.com/sonic-net/sonic-swss/pull/3277 <- for 202311 cherry-pick

**Pull request merged**